src/image_builder.py

```diff
@@ -211,10 +209,34 @@ def image_builder(buildspec, image_types=[], device_types=[]):
         if inference_toolkit_version:
             extra_build_args["SM_TOOLKIT_VERSION"] = inference_toolkit_version
 
+        tag_override = image_config.get("build_tag_override")
+        dockerfile = image_config["docker_file"]
+        target = image_config.get("target")
+        tag_override_regex = r"^(beta|pr):\S+$"
+        if tag_override and build_context == "PR":
+            if is_autopatch_build_enabled(buildspec_path=buildspec):
+                FORMATTER.print("AUTOPATCH ENABLED IN BUILDSPEC, CANNOT OVERRIDE WITH TAG, SORRY!")
+            elif not re.match(tag_override_regex, tag_override):
+                FORMATTER.print(
+                    f"TAG OVERRIDE MUST BE OF FORMAT {tag_override_regex}, but got {tag_override}. Proceeding with regular build."
+                )
+            else:
+                repo_override, t_override = tag_override.split(":")
+                with tempfile.NamedTemporaryFile(mode="w", delete=False) as temp_file_handle:
+                    source_uri = (
+                        f"{image_repo_uri.replace('pr-', f'{repo_override}-')}:{t_override}"
+                    )
+                    temp_file_handle.write(
+                        f"FROM {source_uri}\nLABEL dlc.dev.source_uri={source_uri}"
```

Comment:
Why do we need this label?

Reply:
Technically, we are making a new replica of the beta image, with just the Label as the difference, right? Would it change the SHA of the image and treat it as a new image? What if we have a beta image and a test fails. We the want to run the test for that beta image on the PR and to study it and tweak the test to get a success on the PR. Thereafter, we want to skip the test on the pipeline to allow the image to go to next stage since the test has succeeded on PR. In that case, is it possible that we unknowingly move the tag of the the original beta image to a new image via our PR? We might want to change this behavior if that happens.

Reply:
I guess our tagging mechanism of the new image still uses the same logic as the previous one, so I guess we would not run through this situation. However, just wanted to confirm.

Reply:
Yes, we are basically ferrying over a beta or pr image to a new PR tag, but adding a label to reference the source image. We do not modify existing beta tags in this process - this option is only enable-able in PR context